packages/go_router_builder/example/lib/readme_excerpts.dart

```diff
@@ -0,0 +1,403 @@
+
+  @override
+  Widget build(BuildContext context, GoRouterState state) =>
+      MyScreen(queryParameter: queryParameter);
+}
+// #enddocregion MyRoute
+
+class MyScreen extends StatelessWidget {
+  const MyScreen({required this.queryParameter, super.key});
+  final String queryParameter;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('MyScreen'),
+      ),
+    );
+  }
+}
+
+@TypedGoRoute<PersonRouteWithExtra>(path: '/person')
+// #docregion PersonRouteWithExtra
+class PersonRouteWithExtra extends GoRouteData {
+  PersonRouteWithExtra(this.$extra);
+  final Person? $extra;
+
+  @override
+  Widget build(BuildContext context, GoRouterState state) =>
+      PersonScreen($extra);
+}
+// #enddocregion PersonRouteWithExtra
+
+class PersonScreen extends StatelessWidget {
+  const PersonScreen(this.person, {super.key});
+  final Person? person;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('PersonScreen'),
+      ),
+    );
+  }
+}
+
+// #docregion HotdogRouteWithEverything
+@TypedGoRoute<HotdogRouteWithEverything>(path: '/:ketchup')
+class HotdogRouteWithEverything extends GoRouteData {
+  HotdogRouteWithEverything(this.ketchup, this.mustard, this.$extra);
+  final bool ketchup; // required path parameter
+  final String? mustard; // optional query parameter
+  final Sauce $extra; // special $extra parameter
+
+  @override
+  Widget build(BuildContext context, GoRouterState state) =>
+      HotdogScreen(ketchup, mustard, $extra);
+}
+// #enddocregion HotdogRouteWithEverything
+
+class Sauce {}
+
+class HotdogScreen extends StatelessWidget {
+  const HotdogScreen(this.ketchup, this.mustard, this.extra, {super.key});
+  final bool ketchup; // required path parameter
+  final String? mustard; // optional query parameter
+  final Sauce extra; // special $extra parameter
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Hotdog'),
+      ),
+    );
+  }
+}
+
+// #docregion BookKind
+enum BookKind { all, popular, recent }
+
+class BooksRoute extends GoRouteData {
+  BooksRoute({this.kind = BookKind.popular});
+  final BookKind kind;
+
+  @override
+  Widget build(BuildContext context, GoRouterState state) =>
+      BooksScreen(kind: kind);
+}
+// #enddocregion BookKind
+
+class BooksScreen extends StatelessWidget {
+  const BooksScreen({required this.kind, super.key});
+  final BookKind kind;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('BooksScreen'),
+      ),
+    );
+  }
+}
+
+// #docregion MyMaterialRouteWithKey
+class MyMaterialRouteWithKey extends GoRouteData {
+  static const LocalKey _key = ValueKey<String>('my-route-with-key');
+  @override
+  MaterialPage<void> buildPage(BuildContext context, GoRouterState state) =>
```

Comment:
This, and at least one more below, violate https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#consider-using--for-short-functions-and-methods